### ISSUE TITLE
Skip generated code?

### ISSUE THREAD

**Issue by ainar-g** [shoooooman/go-complexity-analysis#59]
See https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source:

> To convey to humans and machine tools that code is generated, generated source should have a line that matches the following regular expression (in Go syntax):
>
> ```none
> ^// Code generated .* DO NOT EDIT\.$
> ```
>
> This line must appear before the first non-comment, non-blank text in the file.

This feature could also probably be optional and could be disabled with a command-line flag.

**Comment by shoooooman** [comment]
Which part did you mean by generated code?

**Comment by ainar-g** [comment]
@shoooooman, code that is generated automatically by tools.  For example, code generated by Protobuf code generators.

**Comment by shoooooman** [comment]
Yeah, I mean which files do you think should be marked as autogenerated code? I don't think this repo is using protobuf.

**Comment by ainar-g** [comment]
I think there is a misunderstanding.  My proposal is that the tool, `complexity`, ignores (either by default or with a flag) autogenerated code in code bases that it analyses.  Recognizing autogenerated code should be relatively easy, as the overwhelming majority of tools that generate Go code follow the convention documented in the link that I had provided in the original comment.

The motivation is to exclude the results from that code more easily, which is necessary because most of the autogenerated code is not indented to be modified in any way other than just regenerating it.  Currently I exclude them by redirecting stderr to stdout and then filtering it using e.g. `sed`.

**Comment by shoooooman** [comment]
Now I understand your suggestion. Yeah, I also believe that functionality would be helpful. Let me add the flag when I have time, or PR is always welcome.
